api/tests/unit/organisations/test_unit_organisations_tasks.py

```diff
@@ -76,7 +78,56 @@ def test_send_org_over_limit_alert_for_organisation_with_subscription(
     assert kwargs["subject"] == ALERT_EMAIL_SUBJECT
 
 
-def test_finish_subscription_cancellation(db: None):
+def test_subscription_cancellation(db: None) -> None:
+    # Given
+    organisation = Organisation.objects.create()
+    OrganisationSubscriptionInformationCache.objects.create(
+        organisation=organisation,
+    )
+    UserOrganisation.objects.create(
+        organisation=organisation,
+        user=FFAdminUser.objects.create(email=f"{uuid.uuid4()}@example.com"),
+        role=OrganisationRole.ADMIN,
+    )
+
+    assert organisation.subscription_information_cache
+    subscription = organisation.subscription
+    notes = "Notes to be kept"
+    subscription.subscription_id = "id"
+    subscription.subscription_date = timezone.now()
+    subscription.plan = "plan_code"
+    subscription.max_seats = 1_000_000
+    subscription.max_api_calls = 1_000_000
+    subscription.cancellation_date = timezone.now()
+    subscription.customer_id = "customer23"
+    subscription.billing_status = "ACTIVE"
+    subscription.payment_method = "CHARGEBEE"
+    subscription.notes = notes
+
+    subscription.save()
+
+    # When
+    finish_subscription_cancellation()
+
+    # Then
+    organisation.refresh_from_db()
+    subscription.refresh_from_db()
+    assert getattr(organisation, "subscription_information_cache", None) is None
+    assert subscription.subscription_id is None
+    assert subscription.subscription_date is None
+    assert subscription.plan == FREE_PLAN_ID
+    assert subscription.max_seats == MAX_SEATS_IN_FREE_PLAN
+    assert subscription.max_api_calls == MAX_API_CALLS_IN_FREE_PLAN
+    assert subscription.cancellation_date is None
+    assert subscription.customer_id is None
+    assert subscription.billing_status is None
+    assert subscription.payment_method is None
+    assert subscription.cancellation_date is None
+    assert subscription.notes == notes
+
+
+def test_finish_subscription_cancellation(db: None) -> None:
```

Comment:
I think we should add an assert statement here somewhere to make sure the task was scheduled correctly?

Reply:
Yeah that's a good call. Added.